app/services/loader.js

```diff
@@ -97,6 +97,9 @@ export default Service.extend({
       if (parsedResponse) {
         throw parsedResponse;
       }
+      if (response.status === 429) {
```

Comment:
No. Make this a generic implementation such that status code and the original response is always passed in. 


```js
    if (!response.ok) {
      const defaultMessage = httpStatus[response.status];
      const errorResponse = pick(response, ['status', 'ok', 'statusText', 'headers', 'url']);
      errorResponse.statusText = defaultMessage;
      errorResponse.response = parsedResponse;
      errorResponse.errorMessage = isString(parsedResponse) ? parsedResponse : 
        getErrorMessage(
          response.statusText,
          defaultMessage
            ? `${response.status} - ${defaultMessage}`
            : `Could not make ${fetchOptions.type} request to ${fetchOptions.url}`
        );
      throw errorResponse;

    }
```